db-tests/src/test/java/gov/nasa/jpl/aerie/database/ExternalEventTests.java

```diff
@@ -0,0 +1,2188 @@
+        "2024-01-10 00:00:00+00",
+        "2024-08-21 22:36:23.340941+00"
+    );
+    ExternalSource sourceFour = new ExternalSource(
+        "Derivation_Test_03.json",
+        st,
+        dg,
+        "2024-01-21 00:00:00+00",
+        "2024-01-01 12:00:00+00",
+        "2024-01-02 12:00:00+00",
+        "2024-08-21 22:36:28.365244+00"
+    );
+
+    // Second, define the events, spaced by sources 1-4
+    ExternalEvent twoA = new ExternalEvent("2", "DerivationD", "Derivation_Test_00.json", dg, "2024-01-05 23:00:00+00", "01:10:00");
+    ExternalEvent seven = new ExternalEvent("7", "DerivationC", "Derivation_Test_00.json", dg, "2024-01-09 23:00:00+00", "02:00:00");
+    ExternalEvent eight = new ExternalEvent("8", "DerivationB", "Derivation_Test_00.json", dg, "2024-01-10 11:00:00+00", "01:05:00");
+
+    ExternalEvent one = new ExternalEvent("1", "DerivationA", "Derivation_Test_01.json", dg, "2024-01-01 00:00:00+00", "02:10:00");
+    ExternalEvent twoB = new ExternalEvent("2", "DerivationA", "Derivation_Test_01.json", dg, "2024-01-01 12:00:00+00", "02:10:00");
+    ExternalEvent three = new ExternalEvent("3", "DerivationB", "Derivation_Test_01.json", dg, "2024-01-02 23:00:00+00", "03:00:00");
+    ExternalEvent four = new ExternalEvent("4", "DerivationB", "Derivation_Test_01.json", dg, "2024-01-05 21:00:00+00", "03:00:00");
+
+    ExternalEvent five = new ExternalEvent("5", "DerivationC", "Derivation_Test_02.json", dg, "2024-01-05 23:00:00+00", "01:10:00");
+    ExternalEvent six = new ExternalEvent("6", "DerivationC", "Derivation_Test_02.json", dg, "2024-01-06 12:00:00+00", "02:00:00");
+    ExternalEvent twoC = new ExternalEvent("2", "DerivationB", "Derivation_Test_02.json", dg, "2024-01-09 11:00:00+00", "01:05:00");
+
+    ExternalEvent nine = new ExternalEvent("9", "DerivationC", "Derivation_Test_03.json", dg, "2024-01-02 00:00:00+00", "01:00:00");
+
+    // Third, insert types; this can be skipped in the case of multiple calls to upload_source in a given test
+    if (!skip_types) {
+      String[] externalEventTypes = {"DerivationA", "DerivationB", "DerivationC", "DerivationD"};
+      for (String eventType : externalEventTypes) {
+        insertExternalEventType(eventType);
+      }
+      insertExternalSourceType(st);
+    }
+
+    // Fourth, insert derivation group
+    insertDerivationGroup(dg, st);
+
+    // Then, insert sources
+    insertExternalSource(sourceOne);
+    insertExternalSource(sourceTwo);
+    insertExternalSource(sourceThree);
+    insertExternalSource(sourceFour);
+
+    // Finally, insert events
+    insertExternalEvent(twoA);
+    insertExternalEvent(seven);
+    insertExternalEvent(eight);
+    insertExternalEvent(one);
+    insertExternalEvent(twoB);
+    insertExternalEvent(three);
+    insertExternalEvent(four);
+    insertExternalEvent(five);
+    insertExternalEvent(six);
+    insertExternalEvent(twoC);
+    insertExternalEvent(nine);
+  }
+
+  protected void insertStandardTypes() throws SQLException {
```

Comment:
This method is called in every single test in the `DerivedEventsTests` nested class. As it is a `void` method that does not depend on any properties of the tests, these usages ought to be extracted into a `beforeEach` specific to that nested class. (This does not apply to the `ConstraintTest` usages, as it is only used in 4 of those tests).